src/aleph/jobs/job_utils.py

```diff
@@ -28,3 +31,46 @@ def prepare_loop(config_values: Dict) -> Tuple[asyncio.AbstractEventLoop, Config
     init_ipfs_globals(config)
     _ = init_p2p_client(config)
     return loop, config
+
+
+async def perform_db_operations(db_operations: Iterable[DbBulkOperation]) -> None:
+    # Sort the operations by collection name before grouping and executing them.
+    sorted_operations = sorted(
```

Comment:
It's not a matter of efficiency, it's required because it creates a new group for each key change. So if you have a list with group keys in this order: ["a", "b", "a"] it will create 3 groups instead of the 2 we'd like to have.

https://docs.python.org/fr/3.9/library/itertools.html#itertools.groupby

Reply:
OK, makes sense 👍